### PR TITLE
Extra PKCS8 attributes management functions

### PR DESCRIPTION
**Pull request by beldmit**
This PR provides two extra PKCS8 attributes management functions and a basic documentation for the existing and new ones.

##### Checklist

- [x] documentation is added or updated
- [ ] tests are added or updated

### CODE REVIEW

**Comment by t8m** [comment]
Dmitry, you've added a merge commit instead of rebasing.

**Comment by beldmit** [comment]
Thanks! Fix applied.

**Comment by beldmit** [comment]
Ping @openssl/otc 

It's a rather simple PR :)

**Comment by t8m** [comment]
Oh, if you could, please when squashing and merging modify the commit message body to mention the functions being added.

**Comment by slontis** [comment]
Looks like doc-nits is complaining..

File: crypto/asn1/p8_pkey.c [hunk 1]
```diff
@@ -78,3 +78,18 @@ int PKCS8_pkey_add1_attr_by_NID(PKCS8_PRIV_KEY_INFO *p8, int nid, int type,
         return 1;
     return 0;
 }
+
+int PKCS8_pkey_add1_attr_by_OBJ(PKCS8_PRIV_KEY_INFO *p8, const ASN1_OBJECT *obj, int type,
+                                const unsigned char *bytes, int len)
+{
+    if (X509at_add1_attr_by_OBJ(&p8->attributes, obj, type, bytes, len) != NULL)
```
Comment (slontis) [inline crypto/asn1/p8_pkey.c]: return (X509at_add1_attr_by_OBJ(&p8->attributes, obj, type, bytes, len) != NULL);

**Comment by beldmit** [comment]
Will look again tomorrow. It should be the consequence of squash :(

File: doc/man3/PKCS8_pkey_add1_attr.pod [hunk 2]
```diff
@@ -0,0 +1,54 @@
+=pod
+
+=head1 NAME
+
+PKCS8_pkey_get0_attrs, PKCS8_pkey_add1_attr, PKCS8_pkey_add1_attr_by_NID, PKCS8_pkey_add1_attr_by_OBJ - PKCS8 attrs functions
```
Comment (slontis) [inline doc/man3/PKCS8_pkey_add1_attr.pod]: PKCS8 attribute functions

**Comment by slontis** [comment]
I am getting the same error after rebasing a commit - which means master is broken currently.

File: crypto/asn1/p8_pkey.c [hunk 3]
```diff
@@ -78,3 +78,14 @@ int PKCS8_pkey_add1_attr_by_NID(PKCS8_PRIV_KEY_INFO *p8, int nid, int type,
         return 1;
     return 0;
 }
+
+int PKCS8_pkey_add1_attr_by_OBJ(PKCS8_PRIV_KEY_INFO *p8, const ASN1_OBJECT *obj, int type,
+                                const unsigned char *bytes, int len)
+{
+    return (X509at_add1_attr_by_OBJ(&p8->attributes, obj, type, bytes, len) != NULL)
```
Comment (slontis) [inline crypto/asn1/p8_pkey.c]: oops - missing ';' - and same below

**Comment by beldmit** [comment]
@slontis, @t8m - could you please reconfirm?

File: crypto/asn1/p8_pkey.c (hunk 3 above)
Comment (beldmit) [reply crypto/asn1/p8_pkey.c]: Fixed. Thanks!

File: crypto/asn1/p8_pkey.c (hunk 1 above)
Comment (beldmit) [reply crypto/asn1/p8_pkey.c]: Done.

**Comment by beldmit** [comment]
Merged after fixing a conflict in util/libcrypto.num. Thanks!

**Pull request closed without merging**